TACC/Frontera-Portal
# task/FP-213: Shared Workspaces (Projects) Frontend

## Overview: ##

Feature branch for projects frontend

## PR Status: ##


## Related Jira tickets: ##

* [FP-213](https://jira.tacc.utexas.edu/browse/FP-213)

## Summary of Changes: ##

## Testing Steps: ##

0. Add the following to settings_secret:
```
_PORTAL_DATAFILES_STORAGE_SYSTEMS = [
    {
        'name': 'Shared Workspaces',
        'scheme': 'projects',
        'api': 'tapis',
        'icon': None
    }
]
```
1. Set your projects prefix settings (and re-use CEP project root root)
```
_PORTAL_DATA_DEPOT_PROJECTS_SYSTEM_PREFIX = 'frontera.SOMETHINGUNIQUEdev.project'
_PORTAL_PROJECTS_NAME_PREFIX = _PORTAL_DATA_DEPOT_PROJECTS_SYSTEM_PREFIX
_PORTAL_PROJECTS_ID_PREFIX = _PORTAL_NAMESPACE.upper()
_PORTAL_PROJECTS_ROOT_HOST = 'data.tacc.utexas.edu'
_PORTAL_PROJECTS_ROOT_DIR = '/corral-repl/tacc/aci/CEP/projects'
_PORTAL_PROJECTS_ROOT_SYSTEM_NAME = 'cep.project.root
_PORTAL_PROJECTS_PRIVATE_KEY = (
  bunch of private key lines (multi-line format)
)
_PORTAL_PROJECTS_PUBLIC_KEY = (
  bunch of public key lines (multi-line format but probably doesn't matter as much)
)
```
2. Migrate the projects backend with 
```
python manage.py migrate
```
3. Open up a python shell
4. Set up the initial ProjectId model 
```
from portal.apps.projects.models.base import ProjectId
ProjectId.objects.create(value=1).save()
```
5. Create a project using the "+ Add -> Add Shared Workspace" button in Data Files
6. Check to see that you can see your project in the list of Shared Workspaces
7. Ensure file listing, project description updating, title updating, team member updating works


## UI Photos:

## Notes: ##

Todo:
- [x] 'None' description in created projects
- [X] Determine if modals should have horizontal-line (ModalFooter) or not (yes, they should)
- [x] Disable button in edit descriptions modal if there is nothing to update
- [X] ~Create ticket (or resolve) that we prompt users to confirm closing the window if there are unsaved changes (note an undefined ux issue)~ Wes has added a note in a Jira issue about confirmation in modals
- [x] Long titles need to be handled in Shared workspace listing.
- [x] Scrolling when there is lots of workspaces
- [x] Select/Copy Modals: text size in shared-workspace listing 
- [ ] Select/Copy Modals:  spacing between "Workspace Title" and "Destination"
